events.js

```diff
@@ -1,472 +1,488 @@
 'use strict'
 
 module.exports = function (opts) {
+  var exports = {}
 
-var exports = {}
+  var getReceive = opts.getReceive
+  var getReplicate = opts.getReplicate
+  var getSequence = opts.getSequence
 
-var getReceive = opts.getReceive
-var getReplicate = opts.getReplicate
-var getSequence = opts.getSequence
+  exports.note = opts.note
+  exports.getReceive = opts.getReceive
+  exports.getReplicate = opts.getReplicate
+  exports.getSequence = opts.getSequence
 
-exports.note = opts.note
-exports.getReceive = opts.getReceive
-exports.getReplicate = opts.getReplicate
-exports.getSequence = opts.getSequence
-
-function isEmpty (o) {
-  for(var k in o) return false
-  return true
-}
+  function isEmpty (o) {
```

Comment:
while we are at cleaning up: do all these `function`s need to be scoped/closured  inside the `module.exports = (opts) => ...` thing?

Reply:
If they are only used in the file I guess that is fine?

Reply:
Hmm yeah, it is a bit confusing in the other files. Will make a note of that. Thanks